redfish-core/include/utils/power_supply_utils.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "dbus_utility.hpp"
+#include "error_messages.hpp"
+
+#include <boost/system/error_code.hpp>
+#include <sdbusplus/message.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace redfish
+{
+
+namespace power_supply_utils
+{
+
+inline void getInputHistoryPaths(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& validPowerSupplyPath,
+    std::function<void(const std::vector<std::string>& historyPaths)>&&
+        callback)
+{
+    std::string associationPath = validPowerSupplyPath + "/input_history";
+    dbus::utility::getAssociationEndPoints(
+        associationPath, [asyncResp, callback{std::move(callback)}](
+                             const boost::system::error_code& ec,
+                             const dbus::utility::MapperEndPoints& endpoints) {
+            if (ec)
+            {
+                if (ec.value() != EBADR)
+                {
+                    messages::internalError(asyncResp->res);
```

Comment:
A lot times we do  LOG_ERROR with the ec value 
``` BMCWEB_LOG_ERROR << ``` when we have an internalError, it helps with debug. Oh well, I feel these aren't that important
 E.g. 
 https://github.com/ibm-openbmc/bmcweb/blob/1050/redfish-core/lib/account_service.hpp#L409